### ISSUE TITLE
[For 10.1] Dropbox External Storage Release 1.1.0

### ISSUE THREAD

**Issue by pmaier1** [owncloud/files_external_dropbox#10]
# App release template

When marking a task as done, please add your name to the task!

### Reason
- Ironing out migration and better integration with core

#### Still open from v1

- ~~IMPORTANT: write a migration that switches every oc_external_config entry that has the old Dropbox backend to the one from this app~~ => users need to recreate and reauthorize the endpoint anyway, so not needed
- [x] get https://github.com/owncloud/core/pull/29077 merged into core's master and stable10 for upcoming 10.0.4

#### Second version: after 10.0.4 release
- [x] must work with PHP 5.6: https://github.com/owncloud/files_external_dropbox/issues/14
- [x] ~~remove flysystem caching adapter from the files_external_dropbox~~ => https://github.com/owncloud/files_external_dropbox/issues/42
- [x] ~~adjust files_external_dropbox to use core's flysystem caching adapter~~ => https://github.com/owncloud/files_external_dropbox/issues/42
- [x] fixes for file name casings 
- [x] semver set max-version to 10

### QA
- [x] Create **QA Milestone**
- [x] Move items from `development` to `qa` 
- [x] Create release branch "release-1.1.0"
- [x] Create "1.1.0 DropBox Testplan" https://github.com/owncloud/files_external_dropbox/issues/44
- [x] Document Test results and set QA-ready label

### Documentation
- [ ] Changes in the docs needed?

### Marketing
- [ ] Ping RocketChat channel #marketing
- [ ] Check/adapt info.xml stuff
- [ ] Check/adapt screenshot
- [ ] Blog post or other communication if applicable

### Build & Marketplace release

All actions to be done on the release branch from here:
- [x] Bump version in info.xml
- [x] Update CHANGELOG.md - Change CHANGELOG.md 'Unreleased' to '$version - $currentDate' (ex: 3.1.5 - 2018-08-01)
- [x] Create github tag for 'v${version}RC1'
- [x] Build and sign RC tarball from github tag 'v${version}RC1' with `make dist` => https://cloud.owncloud.com/index.php/s/ILlqjBvwsdTqcGW
- [x] Test and document in testplan => Final "Go" from QA (set 'QA-ready' label) => else do another RC after fixing
- [x] Create github tag for final 'v${version}'
- [x] Build and sign final tarball from github tag 'v${version}' with `make dist`
- [x] Smoke test tarball (at least install + enable)
- [x] Github release
- [x] Upload to marketplace
- [x] Publish for customers
- [x] Send mail to release-coordination with link to this ticket and to the changelog
- [x] Rename QA milestone to '$version'
- [x] Add a new 'Unreleased' section in CHANGELOG.md
- [x] Create PR to merge changes back to the `master` branch: https://github.com/owncloud/files_external_dropbox/pull/50

**Comment by PVince81** [comment]
@Hemant-Mann do you have time to work on these ?

**Comment by Hemant-Mann** [comment]
Yes I am a little busy but if you could share the expected timeline for the release of v2 it would help to estimate the time required for the changes

**Comment by Hemant-Mann** [comment]
I think the only major task from the checklist is the Migration

**Comment by patrickjahns** [comment]
I think we could push a v1.0.1 with at least the naming change on the marketplace? so it gets less confusing with two dropbox options

**Comment by PVince81** [comment]
Added "Must work with PHP 5.6". We might want to do another release once this is in place and tested, without waiting for 10.0.4.

**Comment by Hemant-Mann** [comment]
Hey @PVince81 
I was look into Dropbox Webhooks API https://www.dropbox.com/developers/reference/webhooks
But it says that
>The payload of the notification request does not include the actual file changes. It only informs your app of which users have changes.
>
>You will typically want to call /files/list_folder/continue to get the latest changes for each user in the notification, keeping track of the latest cursor for each user as you go.

And we have **only ten seconds to respond to webhook requests**

The solutions mentioned on the website are a) Use threads b) Async Queues
Any Ideas how we can achieve the same in owncloud

**Comment by PVince81** [comment]
PR for changelog to get a view of what has changed since 1.0.1: https://github.com/owncloud/files_external_dropbox/pull/38

we'll look into releasing the app as is

@Hemant-Mann would you have time to remove the flysystem adapter which you moved to core before we do so ? if not we'll leave the app as is, it's ok

**Comment by Hemant-Mann** [comment]
Hi @PVince81 

to be honest it's been quite a time since I last worked on core so it might take some time to do that

**Comment by PVince81** [comment]
@Hemant-Mann yes, I understand. The Dropbox app isn't high on the list of "to release" so this leaves a bit of time. If you don't have time to do it it's also ok.

**Comment by PVince81** [comment]
- [ ] fix packaging: https://github.com/owncloud/files_external_dropbox/issues/40

**Comment by PVince81** [comment]
I've added "verify potential existing issues: https://github.com/owncloud/files_external_dropbox/issues/37" to confirm whether the lowercase issue and others are still happening. Tickets should be raised then.

**Comment by PVince81** [comment]
raised https://github.com/owncloud/files_external_dropbox/issues/42 to address the "flysystem cache adapter" tech debt items. we'll release without that for now

**Comment by davitol** [comment]
<img width="742" alt="screen shot 2018-11-29 at 20 14 11" src="https://user-images.githubusercontent.com/10596560/49248097-3aa43580-f419-11e8-87e3-f377edebdef0.png">


Using the tarball posted above and trying to install it in ownCloud 10.0.10

**Comment by PVince81** [comment]
@davitol the tarball had two issues:
- VCS files were included in signature but excluded later in tar
- tar was truncating long path names resulting in missing file names during signature check...

all fixed in commit https://github.com/owncloud/files_external_dropbox/commit/9fa04e22869b232ea5914067c16fefbfa61a5a4a

please retest with RC2 https://cloud.owncloud.com/index.php/s/KTwSnsPUmccpQpQ

**Comment by davitol** [comment]
> please retest with RC2 https://cloud.owncloud.com/index.php/s/KTwSnsPUmccpQpQ

@PVince81  Tarball is Qa-ready
